### PR TITLE
chore: Log only potential system generated cyclic dependency errors to sentry

### LINKED ISSUE #25687: [POC] Differentiating between user generated cyclic dependency issue vs system generated
Make sure that the entity doesn't depend on its parent node in any case.

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

This PR ensures that only potential system-generated cyclic dependency errors are logged on sentry.

#### PR fixes following issue(s)
Fixes #25687

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/workers/common/DataTreeEvaluator/index.ts [hunk 1]
```diff
@@ -1207,7 +1207,7 @@ export default class DataTreeEvaluator {
 
   sortDependencies(
     dependencyMap: DependencyMap,
-    diffs?: (DataTreeDiff | DataTreeDiff[])[],
```
Comment (arunvjn) [inline app/client/src/workers/common/DataTreeEvaluator/index.ts:1210]: Could you please help me understand why the type was `(DataTreeDiff | DataTreeDiff[])[]` before ?

Comment (ohansFavour) [reply app/client/src/workers/common/DataTreeEvaluator/index.ts:1210]: I am not sure why that was the type initially, `(DataTreeDiff | DataTreeDiff[])[]` has been the type for a long time. After investigation, I noticed that the second argument (diffs) was never passed while invoking the sortDependencies method anywhere in the codebase.

**Review by arunvjn — APPROVED** [review]

**Pull request merged**